src/algebra/group_with_zero/defs.lean

```diff
@@ -45,6 +45,14 @@ mul_zero_class.zero_mul a
 @[ematch, simp] lemma mul_zero (a : M₀) : a * 0 = 0 :=
 mul_zero_class.mul_zero a
 
+lemma pi.single_mul {ι : Type*} [decidable_eq ι] (i j : ι) (a : M₀) (f : ι → M₀) :
+  pi.single i a j * f j = pi.single i (a * f i) j :=
+pi.apply_single (λ i, (* f i)) (λ i, zero_mul _) _ _ _
+
+lemma pi.mul_single {ι : Type*} [decidable_eq ι] (i j : ι) (f : ι → M₀) (a : M₀) :
+  f j * pi.single i a j = pi.single i (f i * a) j :=
```

Comment:
This statement is better unapplied, right?
```suggestion
  f * pi.single i a = pi.single i (f i * a) :=
```

Reply:
Added both versions. In proofs starting with `ext`, the applied version is more useful.